storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -274,7 +279,9 @@ public Optional<String> serverConfigName(String configName) {
             .define(TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_MS, atLeast(-2), MEDIUM,
                 TopicConfig.LOCAL_LOG_RETENTION_MS_DOC)
             .define(TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_BYTES, atLeast(-2), MEDIUM,
-                TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC);
+                TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC)
+            .defineInternal(TopicConfig.REMOTE_LOG_DISABLE_POLICY_CONFIG, STRING, DEFAULT_REMOTE_LOG_DISABLE_POLICY, in("retain", "delete"),
```

Comment:
Could we add a task to publish this config when completed?

Reply:
Yup! Adding one now

Reply:
Created one here - https://issues.apache.org/jira/browse/KAFKA-17031
I will update the rest of this PR in the morning!